internal/service/dhcp/sharednetwork_resource_test.go

```diff
@@ -0,0 +1,2229 @@
+					resource.TestCheckResourceAttr(resourceName, "use_ddns_ttl", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_ddns_update_fixed_addresses", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_ddns_use_option81", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_deny_bootp", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_enable_ddns", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_ignore_client_identifier", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_ignore_dhcp_option_list_request", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_ignore_id", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_lease_scavenge_time", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_logic_filter_rules", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_nextserver", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_options", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_pxe_lease_time", "false"),
+					resource.TestCheckResourceAttr(resourceName, "use_update_dns_on_lease_renewal", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccSharednetworkResource_disappears(t *testing.T) {
+	resourceName := "nios_dhcp_sharednetwork.test"
+	var v dhcp.Sharednetwork
+	name := acctest.RandomNameWithPrefix("sharednetwork")
+	networks := []string{"network/ZG5zLm5ldHdvcmskMjEuMjEuMS4wLzI0LzA:21.21.1.0/24/default",
+		"network/ZG5zLm5ldHdvcmskMjEuMjEuMy4wLzI0LzA:21.21.3.0/24/default"}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckSharednetworkDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSharednetworkBasicConfig(name, networks),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSharednetworkExists(context.Background(), resourceName, &v),
+					testAccCheckSharednetworkDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccSharednetworkResource_Authority(t *testing.T) {
+	var resourceName = "nios_dhcp_sharednetwork.test_authority"
+	var v dhcp.Sharednetwork
+	name := acctest.RandomNameWithPrefix("sharednetwork")
+	networks := []string{"network/ZG5zLm5ldHdvcmskMjEuMjEuNC4wLzI0LzA:21.21.4.0/24/default",
+		"network/ZG5zLm5ldHdvcmskMjEuMjEuNS4wLzI0LzA:21.21.5.0/24/default"}
+	authority := true
+	authorityUpdated := false
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccSharednetworkAuthority(name, networks, authority, true),
```

Comment:
Nitpick :
```suggestion
				Config: testAccSharednetworkAuthority(name, networks, true, true),
```

Since they are used only once , we can just declare them in the call , adding a variable seems unnecessary here !

Reply:
Done